### PR TITLE
DOC: Provide examples of using read_parquet #49739

### PR DESCRIPTION
**Pull request by vvaidy**
- [x] closes #xDOC: Provide examples of using read_parquet #49739
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

@noatamir  @phofl SciPy 2023 sprint

### CODE REVIEW

File: pandas/io/parquet.py [hunk 1]
```diff
@@ -556,7 +556,65 @@ def read_parquet(
...
     -------
     DataFrame
+
+    See Also
+    --------
+    DataFrame.to_parquet : Create a parquet object that serializes a DataFrame.
+
+    Examples
+    --------
+    >>> import pandas as pd
```
Comment (phofl) [inline pandas/io/parquet.py:566]: This line can be removed (it fails a check).

File: pandas/io/parquet.py [hunk 2]
```diff
@@ -556,7 +556,65 @@ def read_parquet(
...
+    which can benefit from multithreading and also potentially be more
+    economical in terms of memory.
+
+    >>> sel = [("foo", ">", 2)]
+    >>> restored_part = pd.read_parquet(BytesIO(df_parquet_bytes), filters=sel)
+    >>> restored_part
+        foo  bar
+    0    3    8
+    1    4    9
+
```
Comment (phofl) [inline pandas/io/parquet.py:615]: No blank line here

**Review by phofl — COMMENTED** [review]
small comments

File: pandas/io/parquet.py (hunk 2 above)
Comment (vvaidy) [reply pandas/io/parquet.py]: Fixed and pushed. Tested locally with  `ci/code_checks.sh docstrings` and seems to check clean

**Comment by mroeschke** [comment]
Can you remove `pandas.read_parquet \` from `code_checks.sh`?

**Comment by vvaidy** [comment]
Hi @mroeschke,

No problem. I did as you requested and reran the tests locally and they succeeded.

I must confess I don't quite understand the implications of that change, but I will look it over later tonight. In the meanwhile, it's done, tested locally and pushed.

Do LMK if you need anything else. Thx.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @vvaidy

**Pull request merged**